discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -25,48 +25,65 @@ def __init__(self, ipfs_peer_host, ipfs_peer_port):
     def get_peer_info(self):
         return self._ipfsid
 
-    def get_metadata_from_json(self, metadata_format, resp_json):
+    def get_metadata_from_json(self, default_metadata_fields, resp_json):
         metadata = {}
-        for parameter, value in metadata_format.items():
+        for parameter, value in default_metadata_fields.items():
             metadata[parameter] = (
                 resp_json.get(parameter) if resp_json.get(parameter) != None else value
             )
         return metadata
 
+    def force_clear_queue_and_stop_task_execution(self, executor):
+        logger.info('IPFSCLIENT | force_clear_queue_and_stop_task_execution - Clearing queue for executor...')
+        executor._threads.clear()
+        concurrent.futures.thread._threads_queues.clear()
+
     # pylint: disable=broad-except
-    def get_metadata(self, multihash, metadata_format, user_replica_set=None):
-        """Retrieve file from IPFS, validating metadata requirements prior to
+    def get_metadata(self, multihash, default_metadata_fields, user_replica_set=None):
+        """Retrieve file from IPFS or gateway, validating metadata requirements prior to
         returning an object with no missing entries
         """
         logger.warning(f"IPFSCLIENT | get_metadata - {multihash}")
-        api_metadata = metadata_format
-        retrieved_from_local_node = False
+        api_metadata = default_metadata_fields
         retrieved_from_gateway = False
+        retrieved_from_local_node = False
+        first_attempt_error = False
```

Comment:
not sure if we need this to track the state of the TPE. if i understand correctly, if the first future errors, you set the variable, if both futures error, you raise.

i would actually not raise at all and just continue because in this line we raise anyway if the metadata was found neither on the ipfs node nor gateway https://github.com/AudiusProject/audius-protocol/blob/ef0e071bd9427110ae4c763d2641337780aa0781/discovery-provider/src/utils/ipfs_lib.py#L97.

Reply:
sounds good! I'll take this out.